modules/match.js

```diff
@@ -14,13 +14,13 @@ import { createRouterObject, createRoutingHistory } from './RouterUtils'
  * Note: You probably don't want to use this in a browser. Use
  * the history.listen API instead.
  */
-function match({ routes, location, ...options }, callback) {
+function match({ history, routes, location, ...options }, callback) {
   invariant(
     location,
     'match needs a location'
   )
 
-  let history = createMemoryHistory(options)
+  history = history ? history : createMemoryHistory(options)
```

Comment:
You can't do this in the assignment because you need `options`. Just `history || createMemoryHistory(options)` is the best I can think of.